### ISSUE TITLE
GeoipCountry() causes server crash on IP addresses from Caribbean Netherlands

### ISSUE THREAD

**Issue by Vauff** [alliedmodders/sourcemod#1094]
# Help us help you
  - [x] I have checked that my issue [doesn't exist yet](https://github.com/alliedmodders/sourcemod/issues).
  - [x] I have tried my absolute best to reduce the problem-space and have provided the absolute smallest test-case possible.
  - [x] I can always reproduce the issue with the provided description below.

# Environment
  * Operating System version: Windows 10 1809 17763.737
  * Game/AppID (with version if applicable): CS:GO/730/740 1.37.1.6
  * Current SourceMod version: 1.9.0.6282
  * Current SourceMod snapshot: 1.10.0.6447
  * Current Metamod: Source snapshot: 1.11.0-dev+1128
  - [x] I have updated SourceMod to the [latest version](https://www.sourcemod.net/downloads.php) and it still happens.
  - [x] I have updated SourceMod to the [latest snapshot](https://www.sourcemod.net/downloads.php?branch=dev) and it still happens.
  - [x] I have updated SourceMM to the [latest snapshot](https://sourcemm.net/downloads.php?branch=dev) and it still happens.

# Description
Running GeoipCountry() on any IP address from Bonaire, Sint Eustatius, and Saba/Caribbean Netherlands causes an instant server crash. I hardcoded an IP in the example code, however it should crash on any IP from [here](https://myip.ms/view/ip_owners/621245/Sscs_Bv.html).

I understand that this may not still be an issue after #914 is merged. But considering this can cause server crashes, it should probably still be fixed.

# Problematic Code (or Steps to Reproduce)
Reproduce by running sm_crashtest
```
#include <sourcemod>
#include <geoip>

public void OnPluginStart() 
{
	RegConsoleCmd("sm_crashtest", Command_CrashTest, "Crash test");
}

public Action Command_CrashTest(int client, int args) 
{
	char countryName[46];

	GeoipCountry("200.71.254.165", countryName, sizeof(countryName));
	PrintToChatAll(countryName);
	return Plugin_Handled;
}
  ```

# Logs
* Please attach in separate files: game output, library logs, kernel logs, and any other supporting information.
* In case of a crash, please attach minidump or dump analyze output.
[crash_4ammfsmxfxjf.zip](https://github.com/alliedmodders/sourcemod/files/3676987/crash_4ammfsmxfxjf.zip)

**Comment by asherkin** [comment]
Thanks for the report, that must've been fun to track down!

Unfortunately at a glance this looks like a bug in the GeoIP library - it appears to be giving back a non-null pointer to garbage.

https://crash.limetech.org/4ammfsmxfxjf

**Comment by gonzalopixart** [comment]
hi :D, i read some documentation for a few hours. 
I think that he is currently obtaining information to obtain the location of the IP ... from a document that if I am not mistaken is a GeoIP.dat, I may be wrong sorry if it is so. I found reading a plugin called geoip2 promises to have a database to perform the calculations (mmdb format) attached repository
https://github.com/Accelerator74/GeoIP2/releases
The one who posted this plugin also recommend using the geoip.inc

**Comment by peace-maker** [comment]
That extension was since merged into Sourcemod #1245 🎉
